app/logic/Mail/SQL/SQLEMail.ts

```diff
@@ -223,19 +223,89 @@ export class SQLEMail {
   static async saveMultiple(emails: Collection<EMail>) {
     let lock = await this.transactionLock.lock();
     try {
-      await (await getDatabase()).run(sql`BEGIN TRANSACTION`);
-      for (let email of emails) {
-        if (!email.subject) {
-          continue;
+      // We need to save all persons before the email transaction
+      // otherwise it will not appear in the database
+      let db = await getDatabase();
+      await db.executeTransaction(() => {
+        for (let email of emails) {
+          if (!email.subject) {
+            continue;
+          }
+          this.saveMainProperties(db, email);
+          this.savePersons(db, email);
         }
-        await this.save(email);
```

Comment:
This saves not only the email and persons, but also attachments and tags etc., which are in different tables.